Simple Incremental/Assets/Scripts/Monobehaviours/ObjectPooler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+/* This class is used for pooling objects and reusing objects for performance
+ * This works best when lots of the same object are spawned multiple times
+ * such as projectiles from a gun */
+public class ObjectPooler : MonoBehaviour
+{
+    public static ObjectPooler instance;
+    public Dictionary<string, Queue<GameObject>> dict = null;
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+            dict = new Dictionary<string, Queue<GameObject>>();
+        }
+        else
+        {
+            Destroy(this);
+        }
+    }
+
+    public GameObject GetPooledObject(GameObject go)
```

Comment:
I'm giving the responsibility to the ObjectPooler to instantiate the objects on the fly.

If you ask the ObjectPooler for an object, it's going to create a new pool for you if it didn't already have one. Currently this one isn't generating a list of pre-pooled objects like yours was before. This is because I added a new PoolableObject script and I hadn't wrapped my brain around it yet. 

I'm going to re-add your variable "objectsToPool" in a little while. I shouldn't have removed that now that I'm reviewing this code again.

Reply:
Yeah the idea was the ObjectsToPool would just be the initial pools objects but not a complete list of pooled objects. So for example you can start up with 50 projectiles spawned on start. Than if later additional were needed they would be added. Also I wanted it so if objects were needed to be pooled at runtime that could happen. So instead of pooling a bunch of enemies as they spawn they just get dynamically pooled as needed.